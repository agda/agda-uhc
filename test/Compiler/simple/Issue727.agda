-- {-# OPTIONS -v tc.lhs.problem:10 #-}
-- {-# OPTIONS --compile --ghc-flag=-i.. #-}
module Issue727 where

open import Common.Prelude renaming (Nat to ℕ)

Sum : ℕ → Set
Sum 0       = ℕ
Sum (suc n) = ℕ → Sum n

sum : (n : ℕ) → ℕ → Sum n
sum 0       acc   = acc
sum (suc n) acc m = sum n (m + acc)

main = putStrLn (natToString (sum 3 0 1 2 3))
