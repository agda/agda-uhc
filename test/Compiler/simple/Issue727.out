EXECUTED_PROGRAM

ret > ExitSuccess
out > 6
out >
